ChangeLog.d/fix-iar-compiler-warnings.txt

```diff
@@ -0,0 +1,2 @@
+Bugfix
+   * Fix IAR compiler warnings.
```

Comment:
We should mention the same issues as in https://github.com/Mbed-TLS/mbedtls/pull/8102, since (as far as I can tell) they were present in 3.4 and are resolved in 3.5.

Reply:
I've just been looking to confirm this, I don't believe #4300 was in 3.4 as the error was in function `ecp_drbg_seed` and I can't see that function or an analog of it existing in 3.4. #7873 was a problem with control flows skipping the declaration of variable `have_rng` in `ecp_mul_mxz` but this isn't in 3.4.

Reply:
Oh, my bad. Thanks for re-checking. Sorry about this.